src/Microsoft.AspNet.Mvc.ModelBinding/Binders/ComplexModelDtoModelBinder.cs

```diff
@@ -22,13 +22,14 @@ public sealed class ComplexModelDtoModelBinder : IModelBinder
             var dto = (ComplexModelDto)bindingContext.Model;
             foreach (var propertyMetadata in dto.PropertyMetadata)
             {
-                    var propertyModelName = ModelBindingHelper.CreatePropertyModelName(
-                            bindingContext.ModelName,
-                            propertyMetadata.BinderModelName ?? propertyMetadata.PropertyName);
+                var propertyModelName = ModelBindingHelper.CreatePropertyModelName(
+                        bindingContext.ModelName,
+                        propertyMetadata.BinderModelName ?? propertyMetadata.PropertyName);
 
-                var propertyBindingContext = new ModelBindingContext(bindingContext,
-                                                                     propertyModelName,
-                                                                     propertyMetadata);
+                var propertyBindingContext = ModelBindingContext.GetChildModelBindingContext(
+                    bindingContext,
+                    propertyModelName,
+                    propertyMetadata);
```

Comment:
A child can be a a normal property or a collection property, the model names differ in both cases, and more over there are complications in creating the model name for collections ( like index names etc ), that logic needs to be outside MBC.